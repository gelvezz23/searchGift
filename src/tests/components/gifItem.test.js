/** @format */
import React from 'react';
import { shallow } from 'enzyme';
import GifItem from '../../components/gifItem';

describe('Pruebas de el componente <gifItem/>', () => {
	const title = 'un titulo';
	const image = 'https://localhost/algo.jpg';
	const wrapper = shallow(<GifItem title={title} image={image} />);

	test('Debe de mostrar el componente correctamente enviando le props', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe de mostrar el componente correctamente', () => {
		const wrapperDos = shallow(<GifItem />);
		expect(wrapperDos).toMatchSnapshot();
	});

	test('El componente debe tener un titulo', () => {
		const p = wrapper.find('p');
		expect(p.text().trim()).toBe(title);
	});

	test('Debe tener una imagen como la que recibe en los props', () => {
		const img = wrapper.find('img');
		expect(img.prop('src')).toBe(image);
		expect(img.prop('alt')).toBe(title);
		// console.log(img.html());
		// forma 1
		// console.log(img.props().src);
		// forma 2
		// console.log(img.prop('src'));
	});

	test('Debe tener una class animate__fadeInDown', () => {
		const div = wrapper.find('div');
		const className = div.prop('className');
		expect(className.includes('animate__fadeInDown')).toBe(true);
	});
});
