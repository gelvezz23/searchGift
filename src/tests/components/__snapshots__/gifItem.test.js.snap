// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de el componente <gifItem/> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <p />
  <img />
</div>
`;

exports[`Pruebas de el componente <gifItem/> Debe de mostrar el componente correctamente enviando le props 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <p>
    un titulo
  </p>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
</div>
`;
